bin/lib/autovoice.js

```diff
@@ -1,145 +1,102 @@
-
-							const cachedItemDataWithMp3 = dataItemsCache.retrieve(fileIdWithoutDuration);
-							if (cachedItemDataWithMp3) {
-								debug('generatePodcast: retrieved from cache: title=', item.title[0], ', voiceId=', itemData.voiceId);
-								return cachedItemDataWithMp3;
-							} else {
-								debug('generatePodcast: item not cached, so about to TTS, title=', item.title[0], ', voiceId=', itemData.voiceId);
-							}
-
-							return tts.mp3( itemData['contentForReading'], itemData.voiceId )
-							.then( mp3Buffer => {
-
-								let itemDataWithMp3 = Object.assign({}, itemData, {
-									duration      : 60, // <-- NB this needs to be calculated
-									mp3Buffer     : mp3Buffer
-								});
-
-								const fileId = [
-									itemDataWithMp3.fileIdWithoutDuration,
-									'duration=' + itemDataWithMp3.duration
-									].join('&');
-
-								itemDataWithMp3['fileId'] = fileId;
-
-								debug('generatePodcast: post tts.mp3 item [ ' + itemDataWithMp3.processingIndex + ' ] mp3Buffer.length=' + mp3Buffer.length );
-
-								dataItemsCache.store(itemDataWithMp3);
-
-								return itemDataWithMp3;
-							});
-						});
-					}
-				})
-
-			debug('num promises=', promises.length);
-
-			return Promise.all(promises).then(items => {
-				debug('in Promise.all');
-				const feed = constructRSS(rssUrl, items);
-				return feed;
-			}, reason => {
-				debug('in Promise.all rejecting:', reason)
-			});
 
-		})
-		.catch(err => {
-			debug(err);
-		})
-		;
+	if (!item.uuid) {
+		return null;
+	} else {
+		return Promise.resolve( item.uuid )
+		.then(uuid => {
+			const itemData = Object.assign({}, item);
+
+			itemData.voiceId         = voiceId;
+			itemData['is-human']     = false;
+			itemData.format          = 'mp3';
+
+			const contentForReading = formatContentForReading.wrapAndProcessItemData( itemData );
+			itemData.contentForReading         = contentForReading;
+			itemData.contentForReadingHashCode = md5(contentForReading);
+
+			const fileIdWithoutDuration = [
+					 'narrator-id=' + itemData.voiceId,
+									'uuid=' + itemData.uuid,
+							'is-human=' + itemData['is-human'],
+								'format=' + itemData.format,
+							'hashcode=' + itemData.contentForReadingHashCode,
+				].join('&');
+
+			itemData.fileIdWithoutDuration = fileIdWithoutDuration;
+
+			const cachedItemDataWithMp3 = dataItemsCache.retrieve(fileIdWithoutDuration);
+			if (cachedItemDataWithMp3) {
+				debug('processItemToMp3: retrieved from cache: title=', itemData.title, ', voiceId=', itemData.voiceId);
+				return cachedItemDataWithMp3;
+			} else {
+				debug('processItemToMp3: item not cached, so about to TTS, title=', itemData.title, ', voiceId=', itemData.voiceId);
+			}
+
+			return tts.mp3( itemData['contentForReading'], itemData.voiceId )
+			.then( mp3Buffer => {
+
+				let itemDataWithMp3 = Object.assign({}, itemData, {
+					duration      : 60, // <-- NB this needs to be calculated
```

Comment:
It could be noted somewhere in here that the Absorber will amend this number if it's incorrect. Although it's preferred that the providers do it.